server/app/services/geo/AddressCandidate.java

```diff
@@ -0,0 +1,52 @@
+package services.geo;
+
+import com.google.auto.value.AutoValue;
+
+/** This class is for setting and getting an address candidate */
+@AutoValue
+public abstract class AddressCandidate {
+  public static Builder builder() {
+    return new AutoValue_AddressCandidate.Builder();
+  }
+  /**
+   * this method returns an address as a single line
+   *
+   * @return String.
+   */
+  public abstract String getAddress();
+
+  /**
+   * this method returns the location object, which conatiins x and y coordinates
+   *
+   * @return AddressLocation
+   */
+  public abstract AddressLocation getLocation();
+
+  /**
+   * this method returns an integer which represents how well the candidate address matches with the
+   * user inputted address
+   *
```

Comment:
suggestion (blocking): mention the range this value can have in the doc

Reply:
sorted descending (mention this)